tw2023_wallet/Feature/ShareCredential/ViewModels/SharingRequestViewModel.swift

```diff
@@ -208,94 +209,70 @@ class SharingRequestViewModel {
             print("\(errorState): Initialization Failed")
             return .failure(errorState)
         }
+        openIdProvider.setSiopAccount(account: account, accountManager: accountManager)
 
-        let publicKey = accountManager.getPublicKey(index: account.index)
-        let privateKey = accountManager.getPrivateKey(index: account.index)
-
-        let keyPair = KeyPairData(publicKey: publicKey, privateKey: privateKey)
-        openIdProvider.setSecp256k1KeyPair(keyPair: keyPair)
-
```

Comment:
The reason why we set the key pair for the ID token here is to take module dependencies into consideration.

accountManager(PairwiseAccount) implements pairwise accounts using HD Keys, but this part is not included in the SIOP protocol. Therefore, we think it is more appropriate to place pairwise-related processing in the ViewModel, which is outside of OpenIdProvider.

Reply:
@ryosuke-wakaba 
Thank you for taking the time to review and comment on the PR.
As you pointed out, I should not bring any process that is not directly related to SIOPv2 into the OpenIdProvider implementation, so reverted to previous implementation.

In this regard, the following is what I was thinking at the time of implementation: 
I had implemented it with the idea of not doing unnecessary key derivation `deriveNode` if no id_token is requested.
(Since only OpenIdProvider knows whether id_token is requested or not, the process got mixed up.)

I realized that there is no particular problem in deriving the key even if the id_token is not requested, so the process was modified.